source/Camera/InfoUtilOther/ZividBenchmark/ZividBenchmark.cpp

```diff
@@ -383,6 +389,60 @@ namespace
         printAssistedCapture3DResults(allDurations);
     }
 
+    
+    template<typename DataType>
+    void CopyDataTime(Zivid::PointCloud pointCloud, DataType copyname)
+    {
+        std::vector<Duration> copyDataDurations;
+
+        const auto beforecopyData = HighResClock::now();
+        const auto tempcopy = pointCloud.copyData<decltype(copyname)>();
+        const auto aftercopyData = HighResClock::now();
+
+        copyDataDurations.push_back(aftercopyData - beforecopyData);
+        
+        std::cout << copyname << std::endl;
+        printcopyDataResults(copyDataDurations);
+    }
+    
+    void benchmarkCopyDataTime(Zivid::Camera &camera)
+    {
+        const Zivid::CaptureAssistant::SuggestSettingsParameters suggestSettingsParameters{
+            Zivid::CaptureAssistant::SuggestSettingsParameters::AmbientLightFrequency::none,
+            Zivid::CaptureAssistant::SuggestSettingsParameters::MaxCaptureTime{ std::chrono::milliseconds{ 1200 } }
+        };
+        
+        const auto settings{ Zivid::CaptureAssistant::suggestSettings(camera, suggestSettingsParameters) };
+        auto frame = camera.capture(settings);
+        auto pointCloud = frame.pointCloud();
```

Comment:
Can you implement the function in the link below and just use it in benchmarkCopyDataTime (it will be a one-liner here).

https://github.com/zivid/zivid-cpp-samples/blob/223e70cc7a2605cfd38035422e1f5f5fd647917d/source/Applications/Advanced/HandEyeCalibration/HandEyeCalibration/HandEyeCalibration.cpp#L64-L72